SYCL/KernelFusion/barrier_local_internalization.cpp

```diff
@@ -0,0 +1,82 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// UNSUPPORTED: cuda || hip
+// REQUIRES: fusion
+
+// Test complete fusion with local internalization and a combination of kernels
+// that require a work-group barrier to be inserted by fusion.
```

Comment:
If there is no barrier inserted and not all work-items execute in a completely synchronized fashion, the computation outcome will not match the expected result. 

The insertion of barriers is also tested with the LIT tests for the kernel fusion passes, which are part of the tests in `sycl-fusion` on `intel/llvm`.

Reply:
I am more concerned about internalization. Does it have any visible effects other than performance?

Reply:
Yes, if internalization happens correctly, the `tmp` buffer will not be written by the device kernels, so it will retain the value it was initialized to (`-1`). I've added a check for that to the test.